Backend/TimeTracker.Api/Controllers/TimerController.cs

```diff
@@ -0,0 +1,206 @@
+                    Message = "Invalid user.",
+                    Success = false
+                };
+            }
+
+            DateTime timeEntryCreationTime = DateTime.Now;
+
+            TimeEntry timeEntry = (await database.TimeEntries
+                .AddAsync(new TimeEntry{
+                    CreatedTime = timeEntryCreationTime,
+                    Notes = timer.Notes,
+                    Project = timer.Project,
+                    LastModified = timeEntryCreationTime,
+                    User = await authHelper.GetCurrentUser(User, database),
+                    Length = (timeEntryCreationTime - timer.StartTime).TotalMinutes,
+                    Day = timeEntryCreationTime
+                })).Entity;
+
+            TimeEntryDTO returnData = new TimeEntryDTO {
+                Id = timeEntry.Id,
+                Length = timeEntry.Length,
+                Notes = timeEntry.Notes,
+                ProjectId = timer.Project.Id,
+                Day = timeEntry.Day
+            };
+
+            database.Timers.Remove(timer);
+
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<TimeEntryDTO> {
+                Success = true,
+                Data = returnData
+            };
+        }
+
+        // returns a TimerDTO for the given timer id
+        [HttpPost]
+        public async Task<GenericResponseDTO<TimerDTO>> GetTimerById(int timerId) {
+            
+            if(User == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            Timer timer = await database.Timers
+                .AsNoTracking()
+                .Include(t => t.User)
+                .Include(t => t.Project)
+                .FirstOrDefaultAsync(t => t.Id == timerId);
+
+            if(timer == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Could not find timer with a matching Id.",
+                    Success = false
+                };
+            }
+
+            if(timer.User.Id != authHelper.GetCurrentUserId(User)) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Timer does not belong to the current user.",
+                    Success = false
+                };
+            }
+
+            return new GenericResponseDTO<TimerDTO> {
+                Data = new TimerDTO{
+                    Id = timer.Id,
+                    StartTime = timer.StartTime,
+                    Notes = timer.Notes,
+                    ProjectId = timer.Project.Id 
+                },
+                Success = true
+            };
+            
+        }
+
+        // returns a list of timerDTOs for each timer belonging to the logged in user
+        [HttpGet]
+        public async Task<GenericResponseDTO<List<TimerDTO>>> GetAllTimers() {
```

Comment:
I think this is fine as a user will most likely never have a TON of timers that would slow down our frontend from loading all their timers in opposed to how the TimeEntry API works where you can get TimeEntries in a specific time frame. If you want you can add that call in as well. If not I am sure this will be fine.